src/main/java/com/radixdlt/crypto/ECKeyPair.java

```diff
@@ -96,12 +97,23 @@ public static ECKeyPair fromSeed(byte[] seed) {
 		}
 
 		try {
-			return fromPrivateKey(Hash.hash256(seed));
+			return fromPrivateKey(HashUtils.sha256(seed));
 		} catch (PrivateKeyException | PublicKeyException e) {
 			throw new IllegalStateException("Should always be able to create private key from seed", e);
 		}
 	}
 
+	/**
+	 * Restore {@link ECKeyPair} instance from given private key by computing corresponding public key.
+	 * @param privateKey ByteCode which contains private key.
+	 * @return A keypair for provided private key
+	 * @throws PrivateKeyException if input byte array does not represent a private key
+	 * @throws PublicKeyException if public key can't be computed for given private key
+	 */
+	public static ECKeyPair fromPrivateKey(HashCode privateKey) throws PrivateKeyException, PublicKeyException {
```

Comment:
This seems like an incorrect interface to me. We don't want to be supporting creating key pairs from arbitrary hash codes as this can quickly become a security liability.